test/Microsoft.TestPlatform.Utilities.UnitTests/StringUtilitiesTests.cs

```diff
@@ -0,0 +1,98 @@
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    [TestClass]
+    public class StringUtilitiesTests
+    {
+        [TestMethod]
+        public void SplitShouldReturnWhenStringisNullOrEmpty()
+        {
+            var argsList = string.Empty.Tokenize(SplitChar, EscapeChar);
+
+            Assert.IsTrue(argsList.IsNullOrEmpty());
+        }
+
+        [TestMethod]
+        public void SplitShouldReturnWhenStringDoesntContainSplitChar()
+        {
+            var data = "foobar";
+            var argsList = data.Tokenize(SplitChar, EscapeChar);
+            var enumerable = argsList as string[] ?? argsList.ToArray();
+
+            Assert.IsTrue(enumerable.Length == 1);
+            Assert.IsTrue(enumerable.First().Equals(data));
+        }
+
+        [TestMethod]
+        public void SplitShouldSplitWhenStringContainsSplitChar()
+        {
+            var data = "foo,bar";
+            var argsList = data.Tokenize(SplitChar, EscapeChar);
+            var enumerable = argsList as string[] ?? argsList.ToArray();
+
+            Assert.IsTrue(enumerable.Length == 2);
+        }
+
+        [TestMethod]
+        public void SplitShouldSplitWhenStringWithSplitCharStartEnd()
+        {
+            var data = ",foo,bar,";
+            var argsList = data.Tokenize(SplitChar, EscapeChar);
+            var enumerable = argsList as string[] ?? argsList.ToArray();
+
+            Assert.IsTrue(enumerable.Length == 4);
+        }
+
+        [TestMethod]
+        public void SplitShouldEscapeSplitCharWhenEscapedCharPresent()
+        {
+            var data = "foo\\,bar";
+            var argsList = data.Tokenize(SplitChar, EscapeChar);
+            var enumerable = argsList as string[] ?? argsList.ToArray();
+
+            Assert.IsTrue(enumerable.Length == 1);
+            Assert.IsTrue(enumerable.First().Equals("foo,bar"));
+        }
+
+        [TestMethod]
+        public void SplitShouldEscapeSplitCharWhenEscapedNonEscapedCharPresent()
+        {
+            var data = "foo\\,,bar";
+            var argsList = data.Tokenize(SplitChar, EscapeChar);
+            var enumerable = argsList as string[] ?? argsList.ToArray();
+            Assert.IsTrue(enumerable.Length == 2);
+            Assert.IsTrue(enumerable.First().Equals("foo,"));
+        }
+
+        [TestMethod]
+        public void SplitShouldSplitWhenOnlySplitCharPresent()
+        {
+            var data = ",";
+            var argsList = data.Tokenize(SplitChar, EscapeChar);
+            var enumerable = argsList as string[] ?? argsList.ToArray();
+
+            Assert.IsTrue(enumerable.Length == 2);
+        }
+
+        [TestMethod]
+        public void SplitShouldNotSplitWhenNoSplitCharPresent()
+        {
+            var data = "foo\\bar";
+            var argsList = data.Tokenize(SplitChar, EscapeChar);
+            var enumerable = argsList as string[] ?? argsList.ToArray();
+
+            Assert.IsTrue(enumerable.Length == 1);
```

Comment:
Should we use SequenceEqual instead or have another assert? just length check isn't good.

Reply:
Added in the cases where we required.